src/me/mrCookieSlime/Slimefun/listeners/BackpackListener.java

```diff
@@ -116,30 +117,34 @@ private void openBackpack(ItemStack sfItem, int size, PlayerInteractEvent e, Pla
 		
 		if (item.getAmount() == 1) {
 			if (Slimefun.hasUnlocked(p, sfItem, true)) {
-				PlayerProfile profile = PlayerProfile.get(p);
-				
-				for (int line = 0; line < item.getItemMeta().getLore().size(); line++) {
-					if (item.getItemMeta().getLore().get(line).equals(ChatColor.translateAlternateColorCodes('&', "&7ID: <ID>"))) {
-						BackpackInventory backpack = profile.createBackpack(size);
-						
-						ItemMeta im = item.getItemMeta();
-						List<String> lore = im.getLore();
-						lore.set(line, lore.get(line).replace("<ID>", p.getUniqueId() + "#" + backpack.getID()));
-						im.setLore(lore);
-						item.setItemMeta(im);
-						break;
-					}
-				}
-				
-				if(!SlimefunPlugin.getUtilities().backpack.containsValue(item)) {
-					PlayerProfile.getBackpack(item).open(p);
-					p.playSound(p.getLocation(), Sound.ENTITY_HORSE_ARMOR, 1F, 1F);
-					SlimefunPlugin.getUtilities().backpack.put(p.getUniqueId(), item);
-				}
-				else SlimefunPlugin.getLocal().sendMessage(p, "backpack.already-open", true);
+				Slimefun.getLocal().sendMessage(p, "messages.opening-backpack");
+				PlayerProfile.get(p, profile -> {
+                    for (int line = 0; line < item.getItemMeta().getLore().size(); line++) {
+                        if (item.getItemMeta().getLore().get(line).equals(ChatColor.translateAlternateColorCodes('&', "&7ID: <ID>"))) {
+                            BackpackInventory backpack = profile.createBackpack(size);
+
+                            setBackpackId(p, item, line, backpack.getID());
+                            break;
+                        }
+                    }
+
+                    if(!SlimefunPlugin.getUtilities().backpack.containsValue(item)) {
+                        PlayerProfile.getBackpack(item).open(p);
+                        p.playSound(p.getLocation(), Sound.ENTITY_HORSE_ARMOR, 1F, 1F);
+                        SlimefunPlugin.getUtilities().backpack.put(p.getUniqueId(), item);
+                    }
+                    else SlimefunPlugin.getLocal().sendMessage(p, "backpack.already-open", true);
+				});
```

Comment:
That indentation has gotten a bit out of hand, hasn't it?

Reply:
Welcome to SF code :D